### ISSUE TITLE
Function in .Net 8 with in-process model failed to startup in Linux container hosted in AKS

### ISSUE THREAD

**Issue by anupamk1506** [Azure/azure-functions-host#10413]
**Is your question related to a specific version? If so, please specify:**
.NET 8
dotnet-inproc
linux Container
AKS
V4

**What language does your question apply to? (e.g. C#, JavaScript, Java, All)**
C#

**Question** 

We migrated the function to .Net 8 with in-process model using the steps mentioned in https://github.com/Azure/azure-functions-host/issues/9951#issuecomment-2174365098.

It is working fine locally on windows machine with Visual studio. We deployed function in AKS with Linux containers using base image  **mcr.microsoft.com/azure-functions/dotnet:4.**

The function is not working in AKS, we checked the pod logs where it shows it has issue with Startup.

![image](https://github.com/user-attachments/assets/cc079e07-1ea9-499a-be88-ba42e2234f95)

 
We checked the application insights, where is shows : _Error configuring services in an external startup class. Could not load file or assembly 'Microsoft.Extensions.Configuration.Abstractions, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system cannot find the file specified._

![image](https://github.com/user-attachments/assets/f1999541-0191-4d03-a103-dd315defd0f6)

 
We tried downgrading the version and removing the package 'Microsoft.Extensions.Configuration.Abstractions. However it didn’t work. And got the same error.

The NuGet package used are :
![image](https://github.com/user-attachments/assets/47d999bf-8379-4bf9-ae26-36e209dd364f)

![image](https://github.com/user-attachments/assets/8acb32b7-7532-4cf6-b389-a4b1b10431d2)

We have added keys in local.setting.json and appsetting.json

![image](https://github.com/user-attachments/assets/0f56ba7e-9cf0-4bf4-8f85-444251c49deb)

The same function is working in .Net6 InProc and we are facing this issue after .Net8 upgrade to InProc.

This is blocker to us for our .Net 8 upgrade. Need help in resolving/investigation of this issue.

Let me know if you need more information.

Thanks!

**Comment by kshyju** [comment]
What is the `HOST_VERSION` environment variable value you are seeing in your container?

**Comment by anupamk1506** [comment]
> What is the `HOST_VERSION` environment variable value you are seeing in your container?

@kshyju Currently we haven't set the HOST_VERSION environment variable in docker file

![image](https://github.com/user-attachments/assets/c9c9b370-8d02-4128-b991-3957ae26803c)

Do we require to add this for inproc function ? As per this thread https://github.com/Azure/azure-functions-docker/issues/1105
HOST_VERSION should be set for isolated function for Open telemetry. Please suggest.

**Comment by kshyju** [comment]
The base image sets it. Run the container and you should be able to inspect the env variables.

**Comment by anupamk1506** [comment]
> The base image sets it. Run the container and you should be able to inspect the env variables.

@kshyju , yes, I can this variable is being set when we build the docker image 

![image](https://github.com/user-attachments/assets/a63dd87b-d41b-4a6e-b6e7-6281b2586772)

At Pod level, I can see the variable HOST_VERSION  is properly set

![image](https://github.com/user-attachments/assets/6045c56f-016a-496a-b356-1802b7a17210)


Do we need to override this? Please suggest

**Comment by kshyju** [comment]
I believe you are using an incorrect base image. @satvu Could you please point the image which supports in-proc net8 ?

**Comment by anupamk1506** [comment]
@kshyju Thanks for update. @satvu Can you please share  image which supports in-proc net8?

**Comment by anupamk1506** [comment]
> I believe you are using an incorrect base image. @satvu Could you please point the image which supports in-proc net8 ?

@kshyju Thanks for update. @satvu Can you please share  image which supports in-proc net8?

**Comment by kshyju** [comment]
Please try `mcr.microsoft.com/azure-functions/dotnet:4-dotnet8.0`

**Comment by KlausHans** [comment]
I have the same problem and unfortunately `mcr.microsoft.com/azure-functions/dotnet:4-dotnet8.0` didn't solved it for me. Any other ideas?
